school_schedule_test.py

```diff
@@ -0,0 +1,29 @@
+from school_schedule import registration_open, free_period
+
+from datetime import datetime
+
+
+class TestFreePeriod:
+    def test_first_day(self):
+        assert free_period(datetime(2022, 9, 7)) == "C"
+
+    def test_known_day(self):
+        assert free_period(datetime(2023, 3, 14)) == "G"
+
+
+class TestRegistrationOpen:
```

Comment:
Currently, all these dates are sometime in the future or on the weekend. What happens if it's a valid day, at a valid time, just in the past? Would this be worth testing?